viewing_party/main.py

```diff
@@ -0,0 +1,227 @@
+#************** WAVE 1 ***************
+
+def create_movie(title, genre, rating):
+    
+    new_movie = {}
+    
+    if type(title) == str and type(genre) == str and type(rating) == float:
+        new_movie["title"] = title 
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+
+        return new_movie
+
+    else: 
+        return None
+
+
+def add_to_watched(user_data, movie):
+    watched_list = user_data["watched"]
+    watched_list.append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = user_data["watchlist"]
+    watchlist.append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    watched_list = user_data["watched"]
+
+    for movie in watchlist:
+
+        if title == movie["title"]:
+            watchlist.remove(movie)
+            watched_list.append(movie)
+    return user_data
+
+# #************** WAVE 2 ***************
+
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    
+    for movie in user_data["watched"]:
+        rating_list.append(movie["rating"])
+
+    thesum = sum(rating_list)
+    avg_rating = thesum/(len(rating_list))
+
+    return avg_rating
+
+def get_watched_avg_rating(user_data):
+    sum = 0.0
+    count = 0
+
+    try:
+        for movie in user_data["watched"]:
+            sum += movie["rating"]
+            count += 1
+        avg_rating = sum/count
+
+    except ZeroDivisionError:
+        avg_rating = 0.0
+    
+    return float(avg_rating)
+
+def get_most_watched_genre(user_data):
+#     #grab all the genres and put into watched_genre_list
+    watched_genre_list = []
+    genre_count = {}
+    most_watched_genre = []
+    current_max = 0
+
+    if not user_data["watched"]:
+        return None
+
+    else:
+        for movie in user_data["watched"]:
+            watched_genre_list.append(movie["genre"])
+    
+#   go through list and count occurences of each genre. store this as key:value pairs in genre_count dictionary
+    
+        for genre in watched_genre_list: 
+            if genre in genre_count:
+                genre_count[genre] = genre_count[genre] +1
+            else:
+                genre_count[genre] = 1
+
+# go through genre_count dictionary and add all the genres with the highest counts into the most_watched_genre list. 
+
+        for val in genre_count.values():
+            if val >= current_max:
+                current_max = val
+    
+        for key, values in genre_count.items():
+            if values == current_max:
+                #winner = key
+                #most_watched_genre.append(winner)
+
+        #return winner
+
+                most_watched_genre.append(key)
```

Comment:
Using more meaningful names than `val` or `key` makes code easier to read:

```suggestion
        for count in genre_count.values():
            if count >= current_max:
                current_max = count
    
        for genre, count in genre_count.items():
            if count == current_max:
                most_watched_genre.append(genre)
```

Cleaning up commented out code also helps with readability.